src/test/groovy/unit/org/pih/warehouse/api/ProductApiTest.groovy

```diff
@@ -0,0 +1,117 @@
+package unit.org.pih.warehouse.api
+
+import io.restassured.path.json.JsonPath
+import io.restassured.specification.RequestSpecification
+import org.testng.Assert
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test
+import spock.lang.Shared
+import spock.lang.Specification;
+
+import static io.restassured.RestAssured.*;
+import io.restassured.response.Response;
+
+class ProductApiTest extends Specification {
+
+    @Shared
+    String baseURI = "http://localhost:8080/openboxes";
+
+    @Shared
+    Map headers = [:]
+
+    @Shared
+    boolean setupInitialized = false
+
+    @Shared
+    String productId
+
+    @Shared
+    String locationId = 1
+
+    void setup() {
+        if (!setupInitialized) {
+            RequestSpecification httpRequest = given().formParam("username", "admin").formParam("password", "password");
+            Response response = httpRequest.post(baseURI + "/auth/handleLogin");
+            response.headers().each {
+                if (it.name == "Set-Cookie") {
+                    headers["Cookie"] = it.value.substring(0, (it.value.indexOf(";")))
+                }
+            }
+            httpRequest = given().header("Cookie", "${headers.Cookie}");
+            response = httpRequest.get(baseURI + "/dashboard/chooseLocation/${locationId}");
+            setupInitialized = true
+        }
+
+    }
+
+    void "read product list"() {
+        when: "/api/products end point called"
+        RequestSpecification httpRequest = given().header("Cookie", "${headers.Cookie}");
```

Comment:
Make sure you pass an Accept header to ensure you get a JSON response.
```
RequestSpecification httpRequest = given().header("Accept", "application/json")
    .header("Cookie", "${headers.Cookie}")
```
Otherwise we get this when running tests
```
io.restassured.path.json.exception.JsonPathException: Failed to parse the JSON document
	at io.restassured.path.json.JsonPath$ExceptionCatcher.invoke(JsonPath.java:930)
	at io.restassured.path.json.JsonPath$4.doParseWith(JsonPath.java:895)
	at io.restassured.path.json.JsonPath$JsonParser.parseWith(JsonPath.java:975)
	at io.restassured.path.json.JsonPath.get(JsonPath.java:201)
	at unit.org.pih.warehouse.api.ProductApiTest.get demand of product(ProductApiTest.groovy:76)
Caused by: groovy.json.JsonException: Lexing failed on line: 1, column: 1, while reading '<', no possible valid JSON value or punctuation could be recognized.
	at groovy.json.JsonLexer.nextToken(JsonLexer.java:87)
	at io.restassured.internal.path.json.ConfigurableJsonSlurper.parse(ConfigurableJsonSlurper.groovy:97)
	at io.restassured.internal.path.json.ConfigurableJsonSlurper.parseText(ConfigurableJsonSlurper.groovy:83)
	at io.restassured.path.json.JsonPath$4$1.method(JsonPath.java:893)
	at io.restassured.path.json.JsonPath$ExceptionCatcher.invoke(JsonPath.java:928)
	... 4 more
```